.gitlab/source_test_junit_upload.yml

```diff
@@ -0,0 +1,16 @@
+source_test_junit_upload:
+  # rules:
+  #   !reference [.on_main_or_release_branch]
```

Comment:
Should we uncomment these lines, or is it fine to run this on all branches?

Reply:
I'm leaving these commented until I do all changes that I've wanted, because otherwise the job won't actually run and I won't be able to test the changes :) I have it on my todo list to uncomment before I merge.

Reply:
I'm not sure what kind of load the CI App supports, but it may be fine to leave this on in all branches?
Having this on all branches may make it easier to detect flaky tests (due to the volume of tests run vs. only reporting on main and release branches), and you can always filter by branch in the UI to only see the state of `main` + release branch tests if needed.

Reply:
TBH I don't feel like the data from branches can be used for anything, since the fact that a test fails on a feature branch doesn't tell you anything - you don't know if it was broken (and then fixed before merging) or if it flaked out. That's why I propose gathering the data from where we're sure it will be meaningful. We can always change this in the future.

Reply:
Let's discuss this in the daily today. I can see how having this on all branches would add "noise" in form of failed tests that were actually intended to fail, but on the other hand it would add so many more runs of all the other tests that actual flakes would stand out regardless.

Reply:
Also, if this is the only thing to discuss from the PR, we can probably merge it now before the code freeze and change that at any time later.

Reply:
Ok, after some more discussion and thinking, I guess we'll never know if there are useful data in feature branches unless we try - so I'll leave this commented for now.